cmd/gen/workflows/flag.go

```diff
@@ -13,22 +13,29 @@ import (
 const (
 	flagCheckSecrets = "check-secrets"
 	flagFlavour      = "flavour"
+	flagArchitecture = "architecture"
 )
 
 type flag struct {
-	CheckSecrets bool
-	Flavours     gen.FlavourSlice
+	CheckSecrets  bool
+	Flavours      gen.FlavourSlice
+	Architectures gen.ArchitectureSlice
 }
 
 func (f *flag) Init(cmd *cobra.Command) {
 	cmd.Flags().VarP(gen.NewFlavourSliceFlagValue(&f.Flavours, gen.FlavourSlice{}), flagFlavour, "f", fmt.Sprintf(`The type of project that you want to generate the workflows for. Possible values: <%s>`, strings.Join(gen.AllFlavours(), "|")))
 	cmd.Flags().BoolVar(&f.CheckSecrets, flagCheckSecrets, true, "If true, also generate a secret-scanning workflow. Possible values: true (default), false.")
+	cmd.Flags().VarP(gen.NewArchitectureSliceFlagValue(&f.Architectures, gen.ArchitectureSlice{gen.ArchitectureDarwin, gen.ArchitectureLinux}), flagArchitecture, "a", fmt.Sprintf(`The architectures to build release artifacts for through workflows. Possible values: <%s>`, strings.Join(gen.AllArchitectures(), "|")))
 }
 
 func (f *flag) Validate() error {
 	if len(f.Flavours) == 0 {
 		return microerror.Maskf(invalidFlagError, "--%s must be one of: %s", flagFlavour, strings.Join(gen.AllFlavours(), ", "))
 	}
 
+	if len(f.Architectures) == 0 {
+		return microerror.Maskf(invalidFlagError, "--%s must contain only values of: %s", flagArchitecture, strings.Join(gen.AllArchitectures(), ", "))
```

Comment:
Why not all of them always?

Reply:
I changed my mind. 2754453 enables all architectures by default :)

Reply:
I don't see why someone would care to disable any of the architectures. This is only for the binaries attached to the release..

Reply:
Just came to mind: so far we were _always_ releasing for darwin and linux. If we _always_ release for darwin then we should _always_ release both amd64 and arm64. Then we can get rid of the flag and the enum and simplify the code here.

Reply:
I’ll create another PR. 👍